### PR TITLE
feat: confirmation modal and additional state checks for alteration handler (HL-1251, HL-1252, HL-1328)

### PR DESCRIPTION
**Pull request by EmiliaMakelaVincit**
## Additional notes :spiral_notepad:

Besides implementing the additional modal functionality, also fixes the minor text issue reported in HL-1328.

### CODE REVIEW

**Review by sirtawast — APPROVED** [review]

File: frontend/benefit/applicant/src/components/applications/alteration/AlterationDeleteModal.tsx [hunk 1]
```diff
@@ -36,8 +36,8 @@ const AlterationDeleteModal = ({
             : 'Suspension'
         }`
       )}
-      submitButtonLabel={t(`common.save`)}
-      cancelButtonLabel={t(`common.backWithoutBack`)}
+      submitButtonLabel=""
```
[the remark this answers is not in the document]
Comment (EmiliaMakelaVincit) [reply frontend/benefit/applicant/src/components/applications/alteration/AlterationDeleteModal.tsx:39]: Yeah, they're completely ignored if the component uses `customContent`, but still required. I noticed a few components using this that had some nonexistent keys defined for them and blanked them instead.

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr3061.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr3061.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr3061.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr3061.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr3061.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**